cron_scripts/password_expiration_notification.pl

```diff
@@ -0,0 +1,290 @@
+    }
+    if ( scalar @data != 0 ) {
+        my $pm = Parallel::ForkManager->new( $opts{max_proc} );
+      ACCOUNTS:
+        foreach my $acc (@data) {
+            $pm->start and next ACCOUNTS;
+            my $mail           = $acc->{mail};
+            my $mobile         = $acc->{mobile};
+            my $displayName    = $acc->{displayName} || 'User';
+            my $expire_in_days = $acc->{'expire_in_days'};
+            if ( $mail ) {
+                my $mail_content = $config->{mail_content};
+                $mail_content =~ s/{displayName}/$displayName/g;
+                $mail_content =~ s/{numDays}/$expire_in_days/g;
+                my $mail_data = {
+                    'to' => $mail,
+                    'subject' =>
+"Your Mail access password expire in $acc->{'expire_in_days'} days",
+                    'body' => $mail_content
+                };
+                if ( my $sendmail =
+                    eval { $soap_api->sendmessage($mail_data) } )
+                {
+                    warn("DEBUG: Mail sent to $mail\n") if ($Debug);
+                }
+                else {
+                    warn( "Error: for $mail on sending mail. Message: "
+                          . ZCS::CustomAPI->Error );
+                }
+            }
+            if ( $mobile && $SendSMS ) {
+                my $sms_content = $config->{sms_content};
+                $sms_content =~ s/{displayName}/$displayName/g;
+                $sms_content =~ s/{numDays}/$expire_in_days/g;
+                my $sms_url =
+"$config->{SMSHOST}?username=$config->{SMSUSER}&pin=$config->{SMSPIN}&mnumber=$mobile&message=$sms_content";
+
+#https://hostname/HttpLink?username=zimbra.sms&pin=redacted&message=hifromnic&mnumber=919922429908&signature=NICSMS
+
+                my $sms_res = LWP::UserAgent->new->get($sms_url);
+
+                #{ "_rc": 500, "_msg": "SMS Sent Successfully" }
+                if ( $sms_res->{'_rc'} != 000 ) {
+                    warn(
+"Error: for $mail on sending SMS. Code: $sms_res->{'_rc'} Message: $sms_res->{'_msg'}"
+                    );
+                }
+                else {
+                    warn("DEBUG: SMS sent to user $mail\n") if ($Debug);
+                }
+            }
+            $pm->finish;
+        }
+        $pm->wait_all_children;
+    }
+    else {
+        warn("DEBUG: No users found!\n") if ($Debug);
+    }
+
+}
+else {
+    pod2usage(
+        -exitval => 1,
+        -msg     => ZCS::CustomAPI->Error
+    );
+}
+
+sub getSoapResponseData {
+    my $searched_result = shift;
+    my $inter           = $searched_result->{'_value'}[0];
+    my $return_attr     = {};
+    foreach my $attrs ( @{ $$inter->{'_value'} } ) {
+        $return_attr->{ $attrs->attr->{'n'} } = $attrs->value;
+    }
+    return $return_attr;
+}
+
+sub process_options {
+    my %opts = (
+        pass_expiration_days => 90,
+        notification_days    => 14,
+		max_proc             => 10
```

Comment:
nit: indentation is off here and several lines below - perltidy can fix this for you